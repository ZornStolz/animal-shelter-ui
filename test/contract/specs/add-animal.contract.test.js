import { provider } from "../config/init-pact";
import { Matchers } from "@pact-foundation/pact";
import { AnimalController } from "../../../controllers";

describe("Animal Service", () => {
  describe("When a request to post an animal is made", () => {
    beforeAll(async () => {
      await provider.setup();
      await provider.addInteraction({
        uponReceiving: "a request to post an animal",
        state: "add animal",
        withRequest: {
          method: "POST",
          path: "/animals",
        },
        willRespondWith: {
          status: 201,
          body: Matchers.eachLike({
            id: Matchers.like(69),
            name: Matchers.like("manchas"),
            breed: Matchers.like("Bengali"),
            gender: Matchers.like("Female"),
            vaccinated: Matchers.boolean(true),
            vaccines: ["Leptospirosis", "Parvovirus"],
          }),
        },
      });
    });

    test("should return the correct data", async () => {
      const animal = {
        name: "manchas",
        breed: "Bengali",
        gender: "Female",
        vaccinated: true,
        vaccines: ["Leptospirosis", "Parvovirus"],
      };
      const response = await AnimalController.register(animal);

      expect(response.data).toMatchSnapshot();
      await provider.verify();
    });

    afterAll(() => provider.finalize());
  });
});
