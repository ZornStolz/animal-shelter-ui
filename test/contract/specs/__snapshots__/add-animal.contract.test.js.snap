// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to post an animal is made should return the correct data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "id": 69,
    "name": "manchas",
    "vaccinated": true,
    "vaccines": Array [
      "Leptospirosis",
      "Parvovirus",
    ],
  },
]
`;
